docs/guides.md

```diff
@@ -0,0 +1,413 @@
+to be _synchronous_ first. This helps us build our suspense mode,
+and it helps your components receive cached data on their initial
+mount without rerendering.
+
+This why **all exchanges should be ordered synchronous first and
+asynchronous last**.
+
+The default order of exchanges is:
+
+```js
+import { dedupExchange, cacheExchange, fetchExchange } from 'wonka';
+
+// Also exported as `defaultExchanges`:
+[dedupExchange, cacheExchange, fetchExchange];
+```
+
+Both the `dedupExchange` and `cacheExchange` are completely
+synchronous and only the `fetchExchange` is asynchronous since
+it makes a `fetch` request and waits for a server response.
+
+When you're adding more exchanges you obviously have a reason
+to put them in a specific order. For instance, an authentication exchange
+needs to go before the `fetchExchange`. And a secondary cache would
+maybe go in front of the default cache exchange.
+
+But to ensure the correct behaviour of suspense mode and
+the initialization of our hooks, it's vital to order your exchanges
+so that synchronous exchanges come first and asynchronous ones
+come last.
+
+## Authentication
+
+Managing and refreshing tokens is a very common case in
+modern application development. In this part we'll build
+this exchange from scratch.
+
+> _Note:_ Setting up a full-scale authentication exchange would be
+> out of scope here. Instead this section teaches the basics on
+> how to wait for an asynchronous request to complete when
+> necessary before letting operations through.
+
+So let's start with a basic template for an exchange
+
+```js
+import { pipe } from 'wonka';
+
+export const refreshTokenExchange = ({ forward }) => {
+  return operations$ => {
+    return pipe(
+      operations$,
+      forward
+    );
+  };
+};
+```
+
+As of now it enters the exchange and tells it to continue due
+to forward being invoked. So this is basically a noop exchange.
+
+Next up is writing some code that refreshes our token, so imagine
+the following method:
+
+> note that these methods should be replaced with your own logic.
+
+```js
+const refreshToken = () => {
+  return Promise.resolve(
+    fetch('/refreshToken', {
+      headers: {
+        'application-type': 'application/json',
+        refreshToken: window.localStorage.get('refreshToken'),
+      },
+    })
+      .then(res => res.json())
+      .then(res => {
+        window.localStorage.setItem('token', res.data.token);
+        return res.data.token;
+      })
+      .catch(console.error)
+  );
+};
+```
+
+Now that we have a way to refresh our token
+we can transform the previous exchange to handle the
+promise that the `refreshToken` function will return.
+
+```js
+import { pipe, fromPromise, map } from 'wonka';
+
+export const refreshTokenExchange = ({ forward }) => {
+  return operations$ => {
+    return pipe(
+      operations$,
+      pipe(
+        fromPromise(refreshToken()),
+        map(newToken => ({ ...op, context: { ...op.context, token: newToken } }))
```

Comment:
what's `op` here?

Reply:
From the mergemap see line 351 or the codesandbox